src/probability/variance.lean

```diff
@@ -0,0 +1,241 @@
+use the latter as the definition, to ensure better behavior even in garbage situations. -/
+def variance {Ω : Type*} {m : measurable_space Ω} (f : Ω → ℝ) (μ : measure Ω) : ℝ :=
+μ[(f - (λ x, μ[f])) ^ 2]
+
+@[simp] lemma variance_zero {Ω : Type*} {m : measurable_space Ω} (μ : measure Ω) :
+  variance 0 μ = 0 :=
+by simp [variance]
+
+lemma variance_nonneg {Ω : Type*} {m : measurable_space Ω} (f : Ω → ℝ) (μ : measure Ω) :
+  0 ≤ variance f μ :=
+integral_nonneg (λ x, sq_nonneg _)
+
+lemma variance_mul {Ω : Type*} {m : measurable_space Ω} (c : ℝ) (f : Ω → ℝ) (μ : measure Ω) :
+  variance (λ x, c * f x) μ = c^2 * variance f μ :=
+calc
+variance (λ x, c * f x) μ
+    = ∫ x, (c * f x - ∫ y, c * f y ∂μ) ^ 2 ∂μ : rfl
+... = ∫ x, (c * (f x - ∫ y, f y ∂μ)) ^ 2 ∂μ :
+  by { congr' 1 with x, simp_rw [integral_mul_left, mul_sub] }
+... = c^2 * variance f μ :
+  by { simp_rw [mul_pow, integral_mul_left], refl }
+
+lemma variance_smul {Ω : Type*} {m : measurable_space Ω} (c : ℝ) (f : Ω → ℝ) (μ : measure Ω) :
+  variance (c • f) μ = c^2 * variance f μ :=
+variance_mul c f μ
+
+localized
+"notation `Var[` X `]` := probability_theory.variance X measure_theory.measure_space.volume"
+in probability_theory
+
+variables {Ω : Type*} [measure_space Ω] [is_probability_measure (volume : measure Ω)]
+
+lemma variance_def' {X : Ω → ℝ} (hX : mem_ℒp X 2) :
+  Var[X] = 𝔼[X^2] - 𝔼[X]^2 :=
+begin
+  rw [variance, sub_sq', integral_sub', integral_add'], rotate,
+  { exact hX.integrable_sq },
+  { convert integrable_const (𝔼[X] ^ 2),
+    apply_instance },
+  { apply hX.integrable_sq.add,
+    convert integrable_const (𝔼[X] ^ 2),
+    apply_instance },
+  { apply integrable.mul_const',
+    apply integrable.const_mul _ 2,
+    exact hX.integrable ennreal.one_le_two },
+  simp only [integral_mul_right, pi.pow_apply, pi.mul_apply, pi.bit0_apply, pi.one_apply,
+    integral_const (integral ℙ X ^ 2), integral_mul_left (2 : ℝ), one_mul,
+    variance, pi.pow_apply, measure_univ, ennreal.one_to_real, algebra.id.smul_eq_mul],
+  ring,
+end
+
+lemma variance_le_expectation_sq {X : Ω → ℝ} :
+  Var[X] ≤ 𝔼[X^2] :=
+begin
+  by_cases h_int : integrable X, swap,
+  { simp only [variance, integral_undef h_int, pi.pow_apply, pi.sub_apply, sub_zero] },
+  by_cases hX : mem_ℒp X 2,
+  { rw variance_def' hX,
+    simp only [sq_nonneg, sub_le_self_iff] },
+  { rw [variance, integral_undef],
+    { apply integral_nonneg,
+      assume a,
+      exact sq_nonneg _ },
+    { assume h,
+      have A : mem_ℒp (X - λ (x : Ω), 𝔼[X]) 2 ℙ := (mem_ℒp_two_iff_integrable_sq
+        (h_int.ae_strongly_measurable.sub ae_strongly_measurable_const)).2 h,
+      have B : mem_ℒp (λ (x : Ω), 𝔼[X]) 2 ℙ := mem_ℒp_const _,
+      apply hX,
+      convert A.add B,
+      simp } }
+end
+
+/-- *Chebyshev's inequality* : one can control the deviation probability of a real random variable
+from its expectation in terms of the variance. -/
+theorem meas_ge_le_mul_variance {X : Ω → ℝ} (hX : mem_ℒp X 2) {c : ℝ} (hc : 0 < c) :
```

Comment:
I don't understand the `mul` in the name. Should it be something like `variance_div_sq` instead?